src/test/kotlin/com/zopa/ktor/opentracing/UtilsTest.kt

```diff
@@ -0,0 +1,130 @@
+package com.zopa.ktor.opentracing
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Test
+
+class UtilsTest {
+    @Test
+    fun `classAndMethodName returns class and method name for method of a class`() = runBlocking<Unit> {
+        var name: String? = null
+
+        class Dog {
+            suspend fun bark() {
+                name = classAndMethodName(this, object {})
+            }
+        }
+
+        Dog().bark()
+
+        assertThat(name).isEqualTo("Dog.bark()")
+    }
+
+    @Test
+    fun `classAndMethodName in init captures class name and does not throw NullPointerException`() {
+        var name: String? = null
+
+        class Dog {
+            init {
+                name = classAndMethodName(this, object {})
+            }
+        }
+
+        Dog()
+
+        assertThat(name).isEqualTo("Dog.()")
+    }
+
+    @Test
+    fun `classAndMethodName does not return function name`() = runBlocking<Unit> {
+        var name: String? = null
+
+        suspend fun bark() {
+            name = classAndMethodName(this, object {})
+        }
+
+        bark()
+
+        assertThat(name).isEqualTo("BlockingCoroutine.invokeSuspend()")
+    }
+
+    @Test
+    fun `classAndMethodName in extension function does not return function name`() = runBlocking<Unit> {
+        var name: String? = null
+
+        fun String.toSomethingElse() {
+            name = classAndMethodName(this, object {})
+        }
+
+        "Hello world".toSomethingElse()
+
+        assertThat(name).isEqualTo("String.invoke()")
+    }
+
+    @Test
+    fun `toPathAndTags with uuid regex replaces and tags uuids`() {
+        val path = "/path/EDF591BA-D318-4F42-8749-4E06F01772BA/view"
+        val tagsToMatch = mapOf(uuidTagAndReplace)
+
+        val pathAndTags = path.toPathAndTags(tagsToMatch)
+
+        assertThat(pathAndTags.path).isEqualTo("/path/<UUID>/view")
+        assertThat(pathAndTags.tags).isEqualTo(mapOf(Pair("UUID", "EDF591BA-D318-4F42-8749-4E06F01772BA")))
+    }
+
+    @Test
+    fun `toPathAndTags with uuid regex returns path unchanged and no tags if no uuids present`() {
+        val path = "/path/view"
+        val tagsToMatch = mapOf(uuidTagAndReplace)
+
+        val pathAndTags = path.toPathAndTags(tagsToMatch)
+
+        assertThat(pathAndTags.path).isEqualTo(path)
+        assertThat(pathAndTags.tags).isEqualTo(emptyMap())
+    }
+
+    @Test
+    fun `toPathAndTags can tag and replace custom strings`() {
+        val path = "/path/12345678-1234/"
+        val tagsToMatch = mapOf(Pair("customId", """[0-9]{8}-[0-9]{4}""".toRegex()))
+
+        val pathAndTags = path.toPathAndTags(tagsToMatch)
+
+        assertThat(pathAndTags.path).isEqualTo("/path/<customId>/")
+        assertThat(pathAndTags.tags).isEqualTo(mapOf(Pair("customId", "12345678-1234")))
+    }
+
+    @Test
+    fun `toPathAndTags can tag and replace multiple strings`() {
```

Comment:
What happens if you match a UUID with a customId?
If you start from
```kotlin
/path/dd988515-936b-4ecd-a6fb-03d74b424f49/hello
```
with
```kotlin
val tagsToMatch = mapOf(
    Pair("customId", """<UUID regex string>""".toRegex()),
)
```
will you get
```kotlin
/path/<customId>/hello`
```

or
```kotlin
/path/<UUID>/hello
```
?

Reply:
I think I'd consider that a config error - you could get either with the current implementation as there's no order 🤷 

I don't really see how you can prevent people from having potentially overlapping regexes.

Are you saying you'd prefer to have an order so that we can say what the expected behaviour would be?

Reply:
I think I would personally prefer having the customId take priority here because it would be a frustrating experience for the user having the default overriding what they specified. Maybe throwing an exception when there is a clash could be an easy way to deal with this?

Reply:
That makes sense. Are you happy with the following?

Config:
```kotlin
install(OpenTracingServer) {
    replaceInPathAndTagSpan(Regex(""), "A")
    replaceInPathAndTagSpan(Regex(""), "B")
}
```

Leading to a priority of:
* A
* B
* UUID

I don't think it's practical to throw an exception for clashes because you can so easily have regexes differ slightly but still perform the same function.

Reply:
Pushed b81e87a